### PR TITLE
Farajaland form updates

### PR DESCRIPTION
**Pull request by tahmidrahman-dsi**
- [x] [Add a custom field to Farajaland death form #6203](https://github.com/opencrvs/opencrvs-core/issues/6203)
- [x] [Add Proof of identity document select to Farajaland forms#6202](https://github.com/opencrvs/opencrvs-core/issues/6202)
- [x] [Add a Spouse page to the Farajaland death form#6204](https://github.com/opencrvs/opencrvs-core/issues/6204)

### CODE REVIEW

File: src/form/birth/index.ts [hunk 1]
```diff
@@ -254,11 +255,16 @@ export const birthForm: ISerializedForm = {
...
             ), // Required field.
-            getNationalID(
-              'informantID',
-              hideIfNidIntegrationEnabled.concat(hideIfInformantMotherOrFather),
-              getNationalIDValidators('informant'),
-              certificateHandlebars.informantNID
+            getIDType(
+              'birth',
+              'informant',
+              hideIfNidIntegrationEnabled.concat(detailsExist),
```
Comment (Zangetsu101) [inline src/form/birth/index.ts]: `detailsExist` doesn't apply to informant so we need `hideIfInformantMotherOrFather` instead

File: src/form/custom-fields.ts [hunk 2]
```diff
@@ -45,3 +48,136 @@ export function createCustomFieldExample(): SerializedFormField {
...
+  required: boolean
+): SerializedFormField {
+  const fieldName: string = `${sectionId}IdNumber${sentenceCase(idValue)}`
+  const validators = getValidators(sectionId, idValue)
+  // eslint-disable-next-line no-console
+  console.log('Custom field addded with handlebar: ', fieldName)
+  return {
+    name: fieldName,
+    required,
+    custom: true,
```
Comment (Zangetsu101) [inline src/form/custom-fields.ts]: Do we need this field to be a custom one? If not then as it's a default field that we support, we need to add it as a valid optional field here: https://github.com/opencrvs/opencrvs-core/blob/develop/packages/config/src/handlers/forms/validation.ts#L228

File: src/form/death/index.ts [hunk 3]
```diff
@@ -61,18 +57,27 @@ import {
   motherFamilyNameConditionals,
   mothersBirthDateConditionals,
   mothersDetailsExistConditionals,
-  spouseDetailsExistConditionals*/
+  spouseDetailsExistConditionals*/,
+  spouseDetailsExistConditionals,
```
Comment (Zangetsu101) [inline src/form/death/index.ts]: Could we clean up the imports here a bit please?

File: src/form/death/index.ts [hunk 4]
```diff
@@ -61,18 +57,27 @@ import {
...
 } from '../common/default-validation-conditionals'
 import { documentsSection, registrationSection } from './required-sections'
 import {
   deceasedNameInEnglish,
   informantNameInEnglish /*,
   fatherNameInEnglish,
   motherNameInEnglish,
-  spouseNameInEnglish*/
+  spouseNameInEnglish*/,
+  spouseNameInEnglish
```
Comment (Zangetsu101) [inline src/form/death/index.ts]: Similarly here

File: src/form/death/index.ts [hunk 5]
```diff
@@ -61,18 +57,27 @@ import {
...
 } from '../common/preview-groups'
 import { certificateHandlebars } from './certficate-handlebars'
 import { getCommonSectionMapping } from '@countryconfig/utils/mapping/field-mapping-utils'
+import { getNumberOfDependants } from './custom-fields'
+import { getIDNumberFields, getIDType } from '../custom-fields'
+import { getSectionMapping } from '@countryconfig/utils/mapping/section/death/mapping-utils'
```
Comment (Zangetsu101) [inline src/form/death/index.ts:83]: and here

**Review by Zangetsu101 — CHANGES_REQUESTED** [review]

File: src/form/custom-fields.ts (hunk 2 above)
Comment (tahmidrahman-dsi) [reply src/form/custom-fields.ts]: Thanks @Zangetsu101, I don't think it needs to be a custom one, but rather an optional field. I am on it

File: src/form/birth/index.ts (hunk 1 above)
Comment (tahmidrahman-dsi) [reply src/form/birth/index.ts]: Good catch @Zangetsu101 I am going to correct that

**Comment by euanmillar** [comment]
Euan to discuss if this should be cherry-picked on top of 1.3.3

**Comment by Zangetsu101** [comment]
@tahmidrahman-dsi Apologies as I don't think we can make the id fields non-custom (i.e. optional fields) as this would restrict us on the types of field we can support, so let's revert that change I requested previously

**Comment by euanmillar** [comment]
@tahmidrahman-dsi we have decided to put this into 1.3.3.  Can you make a new PR based on the release-v1.3.3 branch and cherry-pick these commits onto it and re-submit this PR please?

File: src/client-config.js [hunk 6]
```diff
@@ -23,9 +23,12 @@ window.config = {
   LOGROCKET: '',
   // Use the values in comments when Metabase is running locally
   // http://localhost:4444/public/dashboard/acae0527-74be-4804-a3ee-f8b3c9c8784c#bordered=false&titled=false&refresh=300
-  LEADERBOARDS_DASHBOARD_URL: '',
+  LEADERBOARDS_DASHBOARD_URL:
+    'http://localhost:4444/public/dashboard/acae0527-74be-4804-a3ee-f8b3c9c8784c#bordered=false&titled=false&refresh=300',
```
Comment (Zangetsu101) [inline src/client-config.js]: is this supposed to be here?

Comment (tahmidrahman-dsi) [reply src/client-config.js]: No, removed now

**Review by Zangetsu101 — APPROVED** [review]

**Pull request merged**